core/tasks.py

```diff
@@ -435,6 +434,7 @@ def task():
                     return func(self, *args, **kwargs)
 
             if req:
+                req.is_deferred = True
```

Comment:
That's the wrong point. If you start a deferred task (call the method) the current request will be immediately marked as deferred. This would be wrong for statements after the call.


Short example:
```py
	@exposed
	def test_deferred(self):
		logging.debug("is_deferred: %r (should not)", currentRequest.get().is_deferred)
		self._test_deferred()
		logging.debug("is_deferred: %r (should not)", currentRequest.get().is_deferred)
		
	@tasks.CallDeferred
	def _test_deferred(self):
		logging.debug("I'm a deferred task!")
		logging.debug("is_deferred: %r (should)", currentRequest.get().is_deferred)
```
results into
```
DEBUG    2022-11-22 10:15:04,295 index.py:455] is_deferred: False (should not)
DEBUG    2022-11-22 10:15:04,296 tasks.py:419] make_deferred func=<function Index._test_deferred at 0x7f9755f755a0>, self=<modules.index.Index object at 0x7f975ac24160>, args=(), kwargs={}, queue='default', taskargs={'countdown': None, 'eta': None, 'name': None, 'target': None, 'retry_options': None}
DEBUG    2022-11-22 10:15:04,296 tasks.py:428] func=<function Index._test_deferred at 0x7f9755f755a0> will be executed inline
DEBUG    2022-11-22 10:15:04,296 index.py:457] is_deferred: True (should not)        <---- This is now deferred too
INFO     2022-11-22 10:15:04,404 request.py:358] Running task directly after request: <function Index._test_deferred at 0x7f9755e00670>
DEBUG    2022-11-22 10:15:04,405 index.py:466] I'm a deferred task!
DEBUG    2022-11-22 10:15:04,405 index.py:467] is_deferred: True (should)
[2022-11-22 10:15:04] [200] [GET] /test_deferred
```

---

I suggest to add 
```py
self.is_deferred = True
```
after
https://github.com/viur-framework/viur-core/blob/6b1182d9482dd962332983763cda81f761cf184f/core/request.py#L347